FreeBSD 11.1 compile error with syslog-ng-3.15.1

# syslog-ng

(Please, remove the unnecessary notes, including this one.)

## Version of syslog-ng

```
syslog-ng-3.15.1
```

## Platform

FreeBSD 11.1-RELEASE-p9

# Issue

Compilation of syslog-ng-3.15.1 with same configure paramter from working syslog-ng-3.14.1 on FreeBSD 11.1-RELEASE-p9 fails with the following error:

## Configure parameter
`./configure --prefix=/usr/local/syslog-ng-3.15.1 --enable-dynamic-linking --enable-ipv6 --disable-tcp-wrapper --disable-spoof-source --disable-sql --disable-linux-caps --disable-mongodb --enable-json --disable-smtp --disable-redis --disable-java --disable-python --with-libmongo-client=no PKG_CONFIG=pkgconf OPENSSL_CFLAGS="-I/usr/include/openssl" OPENSSL_LIBS="-L/usr/lib -lcrypto -lssl"`

## gmake output
```
gmake --no-print-directory all-recursive
  CC       modules/native/modules_native_libsyslog_ng_native_connector_a-native-grammar.o
In file included from modules/native/native-grammar.y:26:
In file included from ./modules/native/native-parser.h:28:
In file included from ./lib/cfg-parser.h:29:
In file included from ./lib/cfg-lexer.h:31:
In file included from ./lib/messages.h:29:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.

  CC       modules/native/modules_native_libsyslog_ng_native_connector_a-native-parser.o
In file included from modules/native/native-parser.c:25:
In file included from ./lib/cfg-parser.h:29:
In file included from ./lib/cfg-lexer.h:31:
In file included from ./lib/messages.h:29:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.

  CC       modules/native/modules_native_libsyslog_ng_native_connector_a-parser.o
  AR       modules/native/libsyslog-ng-native-connector.a
  CC       lib/eventlog/src/lib_eventlog_src_libevtlog_la-evtrec.lo
In file included from lib/eventlog/src/evtrec.c:38:
In file included from ./lib/eventlog/src/evt_internals.h:43:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.

  CC       lib/eventlog/src/lib_eventlog_src_libevtlog_la-evtfmt.lo
In file included from lib/eventlog/src/evtfmt.c:44:
In file included from ./lib/eventlog/src/evt_internals.h:43:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.

  CC       lib/eventlog/src/lib_eventlog_src_libevtlog_la-evtout.lo
In file included from lib/eventlog/src/evtout.c:43:
In file included from ./lib/eventlog/src/evt_internals.h:43:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.

  CC       lib/eventlog/src/lib_eventlog_src_libevtlog_la-evtstr.lo
In file included from lib/eventlog/src/evtstr.c:42:
In file included from ./lib/eventlog/src/evt_internals.h:43:
./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.
  CC       lib/eventlog/src/lib_eventlog_src_libevtlog_la-evtctx.lo
In file included from lib/eventlog/src/evtctx.c:43:
In file included from ./lib/eventlog/src/evt_internals.h:43:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^
1 warning generated.

  CC       lib/eventlog/src/lib_eventlog_src_libevtlog_la-evttags.lo
In file included from lib/eventlog/src/evttags.c:42:
In file included from ./lib/eventlog/src/evt_internals.h:43:

./lib/eventlog/src/evtlog.h:150:55: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
                                                      ^

lib/eventlog/src/evttags.c:128:15: error: use of undeclared identifier 'AF_INET'
    inet_ntop(AF_INET, addr, buf, sizeof(buf));
              ^

lib/eventlog/src/evttags.c:136:47: warning: declaration of 'struct in6_addr'
      will not be visible outside of this function [-Wvisibility]
evt_tag_inaddr6(const char *tag, const struct in6_addr *addr)
                                              ^

lib/eventlog/src/evttags.c:136:1: error: conflicting types for 'evt_tag_inaddr6'
evt_tag_inaddr6(const char *tag, const struct in6_addr *addr)
^

./lib/eventlog/src/evtlog.h:150:9: note: previous declaration is here
EVTTAG *evt_tag_inaddr6(const char *tag, const struct in6_addr *addr);
        ^

lib/eventlog/src/evttags.c:141:15: error: use of undeclared identifier
      'AF_INET6'
    inet_ntop(AF_INET6, addr, buf, sizeof(buf));
              ^

2 warnings and 3 errors generated.
gmake[2]: *** [Makefile:11414: lib/eventlog/src/lib_eventlog_src_libevtlog_la-evttags.lo] Error 1
gmake[1]: *** [Makefile:17166: all-recursive] Error 1
gmake: *** [Makefile:7018: all] Error 2
```

--- a/lib/eventlog/src/evtlog.h
+++ b/lib/eventlog/src/evtlog.h
@@ -44,6 +44,11 @@
 # include <syslog.h>
 #endif
 #include <stdarg.h>
+#ifdef __FreeBSD__
+ #include <sys/types.h>
+ #include <sys/socket.h>
+ #include <netinet/in.h>
+#endif
 #include <arpa/inet.h>
 
 #include "evtmaps.h"

--- a/lib/eventlog/src/evtlog.h
+++ b/lib/eventlog/src/evtlog.h
@@ -44,11 +44,9 @@
 # include <syslog.h>
 #endif
 #include <stdarg.h>
-#ifdef __FreeBSD__
- #include <sys/types.h>
- #include <sys/socket.h>
- #include <netinet/in.h>
-#endif
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
 #include <arpa/inet.h>
 
 #include "evtmaps.h"

